node/tests/integration/main.rs

```diff
@@ -382,6 +384,75 @@ async fn test_play_fold() {
     assert_slot_status_fold(&client, target_account_id, game_slot_data).await;
 }
 
+#[tokio::test]
+async fn test_play_check() {
+    let mut client: AzeClient = create_test_client();
+
+    let (game_account, player_account_id, faucet_account_id, game_slot_data) = setup_accounts(
+        &mut client
+    );
+
+    let game_account_storage = game_account.storage();
+
+    let fungible_asset = FungibleAsset::new(faucet_account_id, BUY_IN_AMOUNT).unwrap();
+    let sender_account_id = player_account_id;
+    let target_account_id = game_account.id();
+
+    fund_account(&mut client, sender_account_id, faucet_account_id).await;
+
+    let playcheck_txn_data = PlayCheckTransactionData::new(
+        Asset::Fungible(fungible_asset),
+        sender_account_id,
+        target_account_id
+    );
+
+    let transaction_template = AzeTransactionTemplate::PlayCheck(playcheck_txn_data);
+    let txn_request = client.build_aze_play_check_tx_request(transaction_template).unwrap();
+    execute_tx_and_sync(&mut client, txn_request.clone()).await;
+
+    let note_id = txn_request.expected_output_notes()[0].id();
+    let note = client.get_input_note(note_id).unwrap();
+
+    let tx_template = TransactionTemplate::ConsumeNotes(target_account_id, vec![note.id()]);
+    let tx_request = client.build_transaction_request(tx_template).unwrap();
+    execute_tx_and_sync(&mut client, tx_request).await;
+
+    println!("Executed and synced with node");
+    assert_slot_status_check(&client, target_account_id, game_slot_data.clone(), 1 as u8).await;
+
+    let (player2_account, _) = client
+    .new_game_account(
+        AzeAccountTemplate::PlayerAccount {
+            mutable_code: false,
+            storage_mode: AccountStorageMode::Local,
+        },
+        None
+    )
+    .unwrap();
+
+    fund_account(&mut client, player2_account.id(), faucet_account_id).await;
+
+    let playcheck_txn_data = PlayCheckTransactionData::new(
+        Asset::Fungible(fungible_asset),
+        player2_account.id(),
+        target_account_id
+    );
+
+    let transaction_template = AzeTransactionTemplate::PlayCheck(playcheck_txn_data);
+    let txn_request = client.build_aze_play_check_tx_request(transaction_template).unwrap();
+    execute_tx_and_sync(&mut client, txn_request.clone()).await;
+
+    let note_id = txn_request.expected_output_notes()[0].id();
+    let note = client.get_input_note(note_id).unwrap();
+    
+    let tx_template = TransactionTemplate::ConsumeNotes(target_account_id, vec![note.id()]);
+    let tx_request = client.build_transaction_request(tx_template).unwrap();
+    execute_tx_and_sync(&mut client, tx_request).await;
+
+    println!("Executed and synced with node");
+    assert_slot_status_check(&client, target_account_id, game_slot_data, 2 as u8).await;

```

Comment:
We also need to assert that the `check` procedure successfully updated turn slot for next player turn.